p2p/host/autorelay/options.go

```diff
@@ -75,17 +73,25 @@ type Option func(*config) error
 
 func WithStaticRelays(static []peer.AddrInfo) Option {
 	return func(c *config) error {
-		if c.setMinCandidates {
-			return errStaticRelaysMinCandidates
-		}
 		if c.peerSource != nil {
-			return errStaticRelaysPeerSource
-		}
-		if len(c.staticRelays) > 0 {
-			return errors.New("can't set static relays, static relays already configured")
+			return errAlreadyHavePeerSource
 		}
-		c.minCandidates = len(static)
-		c.staticRelays = static
+
+		WithPeerSource(func(ctx context.Context, numPeers int) <-chan peer.AddrInfo {
+			if len(static) < numPeers {
+				numPeers = len(static)
+			}
```

Comment:
The documentation for `WithPeerSource` says:
> Implementations must send *at most* numPeers, and close the channel when they don't intend to provide any more peers.

Reply:
We can use `WithMaxCandidaters` to make sure that the `relayFinder` is asking us for enough peers. We should probably also add some documentation that there's little point in providing a huge number of static relays.

Reply:
This check _is_ to make sure we send at most numPeers. If we don’t have numPeers we’ll only send what we have. Essentially the ‘min(len(static),numPeers))’. I debated using with ‘WithMaxCandidaters’, but your right I’ll add it.

Reply:
For some reason I thought the `if` was the other way round (`>` instead of `<`). Sorry for that!